File: pwgen.pl
#!/usr/bin/perl

$maxlen = $ARGV[0];
if ($#ARGV > 0) {
    $numpws = $ARGV[1];
} else {
    $numpws = 1;
}

open(SRC, "</dev/urandom") or die "`urandom` not found.";

for($pwcount = 0; $pwcount < $numpws; $pwcount++) {
    $pw = "";
    while (length($pw) < $maxlen) {
        $c = getc SRC;
        if ($c =~ m/[\w\Q!@$&*?\E]/) {
            $pw = $pw . $c;
        }
    }
    print "$pw\n";
}

File: pwgen.py
#!/usr/bin/python
from sys import argv, exit

try:
    src = open("/dev/urandom")
except IOError:
    print "`urandom` device not found."
    print "Either you are not on a Unix-like OS,",
    print "or you have severely messed something up."
    print exit(1)

validchars = ['abcdefghijklmnopqrstuvwxyz',
              'ABCDEFGHIJKLMNOPQRSTUVWXYZ',
              '1234567890',
              '_+=-?!@$&*']

if len(argv) > 2:
    n = int(argv[2])
else:
    n = 1

for arg in xrange(n):
    pw = ""
    while len(pw) < int(argv[1]):
        c = src.read(1)
        if c in ''.join(validchars): pw += c
    print pw
